db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -7183,8 +7183,7 @@ public final TupleDescriptor getDescriptorViaIndex(int indexId,
             rowLocations.add((RowLocation) indexRow1.getColumn(
                     indexRow1.nColumns()).cloneValue(true));
         }
-        heapCC.batchFetch(rowLocations,outRows,null);
-        if (outRows.size() != i) {
+        if (!heapCC.batchFetch(rowLocations,outRows,null)) {
             if(SanityManager.DEBUG){
```

Comment:
It'd be better to substitute this SanityManager block for something else that raises an error